dfaker.py

```diff
@@ -0,0 +1,387 @@
+from datetime import datetime, timedelta, tzinfo
+import time
+from pytz import timezone
+import pytz
+import uuid
+import json
+import argparse 
+import sys
+import cbg_equation
+import statsmodels.api as sm
+
+import random
+import numpy as np 
+
+params = {
+	'year' : 2015, #default datetime settings
+	'month' : 1,
+	'day' : 1,
+	'hour' : 0,
+	'minute' : 0,
+	'zone' : 'US/Pacific', #default zone
+	'num_days' : 30, #default days to generate data for
+	'file' : 'device-data.json', #default json file name 
+	'minify' : False, #compact storage option false by default 
+	'gaps' : False, #randomized gaps in data, off by default 
+	'smbg_freq' : 6 #default fingersticks per day
+}
+
+def parse(args, params):
+	if args.date:
+		try:
+			params['year'] = int(args.date[0:4])
+			params['month'] = int(args.date[5:7]) 
+			params['day'] = int(args.date[8:])
+		
+			if (params['month'] not in range(1, 13) 
+					or params['day'] not in range(1, 32) 
+					or params['year'] not in range(1900, 2100) 
+					or len(args.date) != 10):
+				print('Invalid range: {:s}. Please enter date as YYYY-MM-DD'.format(args.date))
+				sys.exit(1)
+		except:		
+			print('Wrong date format: {:s}. Please enter date as YYYY-MM-DD'.format(args.date))
+			sys.exit(1)
+
+	if args.time:
+		try:
+			params['hour'] = int(args.time[:2])
+			params['minute'] = int(args.time[3:])
+
+			if (params['hour'] not in range(0, 25) 
+					or params['minute'] not in range(0, 61) 
+					or len(args.time) != 5):
+				print('Invalid range: {:s}. Please enter time as HH:MM'.format(args.time))
+				sys.exit(1)
+		except:
+			print('Wrong time format: {:s}. Please enter time as HH:MM'.format(args.time))
+			sys.exit(1)
+	 
+	if args.zone:
+		if args.zone not in pytz.common_timezones:
+			print('Unrecongznied zone: {:s}'.format(args.zone))
+			sys.exit(1)
+		params['zone'] = args.zone
+
+	if args.num_days:
+		params['num_days'] = args.num_days
+
+	if args.file:
+		if args.file[-5:] != '.json':
+			print('Output file must be in be a json file')
+			sys.exit(1)
+		params['file'] = args.file
+
+	if args.minify:
+		params['minify'] = True
+
+	if args.gaps:
+		params['gaps'] = True
+
+	if args.smbg_freq:
+		if smbg_freq == 'high':
+			params['smbg_freq'] = 8
+		elif smbg_freq == 'average':
+			params['smbg_freq'] = 6
+		elif smbg_freq == 'low':
+			params['smbg_freq'] = 3
+		else:
+			print('Invalid frequency: {:s}. Please choose between high, average and low'.format(args.smbg_freq))
+			sys.exit(1)
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-tz', '--timezone', dest='zone', help='Local timezone')
```

Comment:
In a lot of CLI tools, the short forms of flags (e.g., `-tz` is the shortform for `--timezone`) can be combined together. That is, if you had a `-t` flag and a `-z` flag, using `-tz` would apply both. (I can't remember if argparse works like this by default, but it might, at least when the flags are true/false rather than taking args.) So basically short form flags should *always* be one letter. Try changing this one to just `-z` and the `-nd` one below to `-n`?